reconcile/blackbox_exporter_endpoint_monitoring.py

```diff
@@ -166,19 +167,24 @@ def run(dry_run: bool, thread_pool_size: int, defer=None) -> None:
         for p in desired_endpoints
         if p.blackboxExporter
     }
-    ri, oc_map = ob.fetch_current_state(
-        namespaces.values(),
-        thread_pool_size=thread_pool_size,
-        integration=QONTRACT_INTEGRATION,
-        integration_version=QONTRACT_INTEGRATION_VERSION,
-        override_managed_types=["Probe"]
-    )
-    defer(oc_map.cleanup)
-
-    # reconcile
-    for provider, endpoints in desired_endpoints.items():
-        fill_desired_state(provider, endpoints, ri)
-    ob.realize_data(dry_run, oc_map, ri, thread_pool_size)
-
-    if ri.has_error_registered():
-        sys.exit(1)
+
+    if namespaces:
+        ri, oc_map = ob.fetch_current_state(
+            namespaces.values(),
+            thread_pool_size=thread_pool_size,
+            internal=internal,
+            use_jump_host=use_jump_host,
+            integration=QONTRACT_INTEGRATION,
+            integration_version=QONTRACT_INTEGRATION_VERSION,
+            override_managed_types=["Probe"]
+        )
+        defer(oc_map.cleanup)
+
+        # reconcile
+        for provider, endpoints in desired_endpoints.items():
+            fill_desired_state(provider, endpoints, ri)
+        ob.realize_data(dry_run, oc_map, ri, thread_pool_size,
+                        recycle_pods=False)
+
+        if ri.has_error_registered():
+            sys.exit(1)
```

Comment:
Is a registered error also logged somewhere before exit?

Reply:
Ah ok [looks like](https://github.com/app-sre/qontract-reconcile/blob/master/reconcile/utils/openshift_resource.py#L484-L492) a registered error is anyways just a boolean flag - not an error object. So anyways not a concern at this stage :)

Reply:
the errors related to `ResourceInventory a.k.a. ri` are usually logged where they occur, e.g. [here](https://github.com/app-sre/qontract-reconcile/blob/407eda78e4b326396a316b969bd7af149ed7a776/reconcile/openshift_base.py#L469-L474)